tests/test/transport/test_message_endpoint_client.cpp

```diff
@@ -0,0 +1,224 @@
+
+    // Close all endpoint clients
+    REQUIRE_NOTHROW(cli.close(false));
+    REQUIRE_NOTHROW(secondCli.close(true));
+
+    // Close message context
+    context.close();
+}
+
+TEST_CASE("Test send/recv one message", "[transport]")
+{
+    // Get message context
+    auto& context = getGlobalMessageContext();
+
+    // Open the source endpoint client, don't bind
+    SendMessageEndpoint src(thisHost, testPort);
+    src.open(context);
+
+    // Open the destination endpoint client, bind
+    RecvMessageEndpoint dst(testPort);
+    dst.open(context);
+
+    // Send message
+    std::string expectedMsg = "Hello world!";
+    uint8_t msg[expectedMsg.size()];
+    memcpy(msg, expectedMsg.c_str(), expectedMsg.size());
+    REQUIRE_NOTHROW(src.send(msg, expectedMsg.size()));
+
+    // Receive message
+    Message recvMsg = dst.recv();
+    REQUIRE(recvMsg.size() == expectedMsg.size());
+    std::string actualMsg(recvMsg.data(), recvMsg.size());
+    REQUIRE(actualMsg == expectedMsg);
+
+    // Close endpoints
+    src.close();
+    dst.close();
+
+    // Close message context
+    context.close();
+}
+
+TEST_CASE("Test await response", "[transport]")
+{
+    // Get message context
+    auto& context = getGlobalMessageContext();
+
+    // Prepare common message/response
+    std::string expectedMsg = "Hello ";
+    std::string expectedResponse = "world!";
+
+    std::thread senderThread([&context, expectedMsg, expectedResponse] {
+        // Open the source endpoint client, don't bind
+        MessageEndpointClient src(thisHost, testPort);
+        src.open(context);
+
+        // Send message and wait for response
+        uint8_t msg[expectedMsg.size()];
+        memcpy(msg, expectedMsg.c_str(), expectedMsg.size());
+        src.send(msg, expectedMsg.size());
+
+        // Block waiting for a response
+        Message recvMsg = src.awaitResponse(testReplyPort);
+        assert(recvMsg.size() == expectedResponse.size());
+        std::string actualResponse(recvMsg.data(), recvMsg.size());
+        assert(actualResponse == expectedResponse);
+
+        src.close();
+    });
+
+    // Receive message
+    RecvMessageEndpoint dst(testPort);
+    dst.open(context);
+    Message recvMsg = dst.recv();
+    REQUIRE(recvMsg.size() == expectedMsg.size());
+    std::string actualMsg(recvMsg.data(), recvMsg.size());
+    REQUIRE(actualMsg == expectedMsg);
+
+    // Send response, open a new endpoint for it
+    SendMessageEndpoint dstResponse(thisHost, testReplyPort);
+    dstResponse.open(context);
+    uint8_t msg[expectedResponse.size()];
+    memcpy(msg, expectedResponse.c_str(), expectedResponse.size());
+    dstResponse.send(msg, expectedResponse.size());
+
+    // Wait for sender thread
+    if (senderThread.joinable()) {
+        senderThread.join();
+    }
+
+    // Close receiving endpoints
+    dst.close();
+    dstResponse.close();
+
+    // Close message context
+    context.close();
+}
+
+TEST_CASE("Test send/recv many messages", "[transport]")
```

Comment:
These tests could also have a `MessageContextFixture` which sets up and tears down the message context. Might be overkill as it's only a couple of lines in each place, but you get the picture.